modules/gapi/src/executor/gstreamingexecutor.cpp

```diff
@@ -400,53 +401,78 @@ bool QueueReader::getInputVector(std::vector<Q*> &in_queues,
-                in_queues[id] = nullptr;
-                in_constants.resize(in_queues.size());
-                in_constants[id] = std::move(stop.cdata);
-
-                // NEXT time (on a next call to getInputVector()), the
-                // "q==nullptr" check above will be triggered, but now
-                // we need to make it manually:
-                isl_inputs[id] = in_constants[id];
-            }
-            else
+            case Cmd::index_of<cv::GRunArg>():
+                isl_inputs[id] = cv::util::get<cv::GRunArg>(m_cmd[id]);
+                break;
+            case Cmd::index_of<Stop>():
             {
-                GAPI_Assert(stop.kind == Stop::Kind::HARD);
-                rewindToStop(in_queues, id);
-                // After queues are read to the proper indicator,
-                // indicate end-of-stream
-                return false;
-            } // if(Cnst)
-        } // if(Stop)
+                const auto &stop = cv::util::get<Stop>(m_cmd[id]);
+                if (stop.kind == Stop::Kind::CNST)
+                {
+                    // We've got a Stop signal from a const source,
+                    // propagated as a result of real stream reaching its
+                    // end.  Sometimes these signals come earlier than
+                    // real EOS Stops so are deprioritized -- just
+                    // remember the Const value here and continue
+                    // processing other queues. Set queue pointer to
+                    // nullptr and update the const_val vector
+                    // appropriately
+                    m_finishing = true;
+                    in_queues[id] = nullptr;
+                    in_constants.resize(in_queues.size());
+                    in_constants[id] = std::move(stop.cdata);
+
+                    // NEXT time (on a next call to getInputVector()), the
+                    // "q==nullptr" check above will be triggered, but now
+                    // we need to make it manually:
+                    isl_inputs[id] = in_constants[id];
+                }
+                else
+                {
+                    GAPI_Assert(stop.kind == Stop::Kind::HARD);
+                    rewindToStop(in_queues, id);
+                    // After queues are read to the proper indicator,
+                    // indicate end-of-stream
+                    return cv::gimpl::StreamMsg{cv::gimpl::EndOfStream{}};
+               } // if(Cnst)
+               break;
+           }
+           case Cmd::index_of<cv::gimpl::Exception>():
+           {
+               exception =
+                   cv::util::make_optional(cv::util::get<cv::gimpl::Exception>(m_cmd[id]));
+               break;
+           }
+           default:
+               GAPI_Assert(false && "Unsupported cmd type in getInputVector()");
+        }
     } // for(in_queues)
 
+    if (exception.has_value()) {
+        return cv::gimpl::StreamMsg{exception.value()};
+    }
```

Comment:
So if multiple islands throw, you'll return only one, right?
I believe this behavior may be extended in the future, at least worth adding to the backlog.

Reply:
Right, the point there isn't that to expose all exceptions from graph, but to let them be thrown from single main thread.